doc/md/fundamentals/1-basic-syntax/5-characters-strings.md

```diff
@@ -18,41 +18,63 @@ let symbol : Char = '✮';
 
 You can convert a `Text` value `t` to a `Char` iterator using either `t.chars()` or, equivalently, `Text.toIter(t)`.
 
-```motoko no-repl
+:::info [Iter](https://internetcomputer.org/docs/motoko/base/Iter)
+An `Iter<T>` is an object that sequentially produces values of specified type `T` until no more values remain.
+:::
+
+```motoko
 import Text "mo:base/Text";
 import Iter "mo:base/Iter";
 import Char "mo:base/Char";
 
-actor {
-  // Internal function that works with iterators
+persistent actor Alternator {
+
+  // Turn text into an iterator of Char
   func textToChars(t: Text) : Iter.Iter<Char> {
-    t.chars() 
+    t.chars();
   };
-  
-  // Public function that demonstrates using the iterator
-  public func demonstrateTextToChars(t: Text) : async Text {
-    // Use Text.translate for more efficient character transformation
-    return Text.translate(t, func(c) {
-      // This function is applied to each character
-      Char.toText(c)
+
+  // Alternate capitalization
+  public func alternateCaps(t: Text) : async Text {
+    let chars = textToChars(t);
+    var index = 0;
+    // Apply a case function to each char
+    let modified = Iter.map<Char, Text>(chars, func(c: Char) : Text {
+      let charAsText = Char.toText(c);
+      let transformedText = 
+        if (index % 2 == 0) {
+          Text.toUppercase(charAsText)
+        } else {
+          Text.toLowercase(charAsText)
+        };
+      index += 1;
+      transformedText;
     });
+
+    return Text.join("", modified);
   };
-}
+};
 ```
 
+:::note Conversions
+
+- `Char` can be converted to a single-character `Text` using `Text.fromChar(c)`.
+- `Text` can be converted to a single-character `Char` using `Char.toText(t)`.
+:::
+
 ## Strings
 
-Sequences of characters are handled using the [`Text`](https://internetcomputer.org/docs/motoko/base/Text) type, which represents immutable strings of unicode characters delimited with a double quotation mark (`"`).
+Sequences of characters are handled using the [`Text`](https://internetcomputer.org/docs/motoko/base/Text) type, which represents immutable strings of Unicode characters delimited with a double quotation mark (`"`).
 
-```motoko no-repl
+```motoko
 let greeting : Text = "Hello, world!";
 
 // Concatenating text
 
```

Comment:
````suggestion
```

The `#` operation concatenates two `Text` values:

``` motoko
// Concatenating text

````

(split into two displays)